app/models/semester.rb

```diff
@@ -10,6 +10,17 @@ def initialize(semester)
   def self.from_year(year, autumn)
     Semester.new(year + (autumn ? '1' : '2'))
   end
+  def self.current
+    if (1..8).include? Time.current.month
+      if Time.current.month ==1
+        return Semester.new((Time.current.year-1).to_s  + '1')
+      else
+        return Semester.new((Time.current.year-1).to_s + '2')
+      end
+    else
+      return Semester.new((Time.current.year).to_s  + '1')
```

Comment:
Just a code style issue, but we generally put a whitespace around mathematical expressions like `==`, `+` or `-` consistently.

Reply:
Also, you can omit the `return` keyword in the if statements as well since they're going to be implicitly returned.